### PR TITLE
Add support for passing list as cmap in mpl

### PR DESCRIPTION
**Pull request by philippjfr**
As the title says this allows passing a list of colors as a cmap for mpl matching the bokeh behavior and addressing https://github.com/ioam/holoviews/issues/1885.

### CODE REVIEW

**Comment by philippjfr** [comment]
Ready to merge once tests pass.

File: holoviews/plotting/mpl/element.py [hunk 1]
```diff
@@ -694,7 +694,13 @@ def _norm_kwargs(self, element, ranges, opts, vdim):
...
 
         # Define special out-of-range colors on colormap
-        cmap = copy.copy(plt.cm.get_cmap(opts.get('cmap')))
+        cmap = opts.get('cmap')
+        if isinstance(cmap, list):
+            cmap = mpl_colors.ListedColormap(cmap)
+        elif isinstance(cmap, util.basestring):
+            cmap = copy.copy(plt.cm.get_cmap(cmap))
+        else:
+            cmap = copy.copy(cmap)
```
Comment (jlstevens) [inline holoviews/plotting/mpl/element.py]: What is all the copying about?

Comment (philippjfr) [reply holoviews/plotting/mpl/element.py]: I modify the colormaps to set the NaN, over and under colors and we definitely don't want to modify the default colormaps or user supplied ones.

**Comment by jlstevens** [comment]
Looks good. Merging.

**Pull request merged**